pkg/plugins/resources/k8s/native/api/v1alpha1/zz_generated.system.go

```diff
@@ -190,7 +190,7 @@ func (cb *ZoneInsight) SetSpec(spec core_model.ResourceSpec) {
 		panic(fmt.Sprintf("unexpected protobuf message type %T", spec))
 	}
 
-	cb.Spec = &apiextensionsv1.JSON{Raw: util_proto.MustMarshalJSON(s)}
+	cb.Spec = &apiextensionsv1.JSON{Raw: util_proto.MustMarshalJSONSorted(s)}
```

Comment:
Dumb question but using `json.Marshal(s)` just doesn't work here?

Reply:
I've tested it but unfortunately, order is not correct, it preservers proto order

Reply:
Would it be less hacky to do roughly what #11327 does but instead of doing the compare in the main controller code doing it in [PodToDataplane](https://github.com/kumahq/kuma/blob/c82b1190303912344b29b20051d09946238ec304/pkg/plugins/runtime/k8s/controllers/pod_converter.go#L36-L51) ? 
Doing something like:

```
	dataplane.Mesh = util_k8s.MeshOfByAnnotation(pod, ns)
	dataplaneProto, err := p.dataplaneFor(ctx, pod, services, others)
	if err != nil {
		return err
	}
        curSpec := datapane.GetSpec()
        if proto.DeepEqual(curSpec, datapalneProto) {
             return UnchangedError // Or something different.
        }
        dataplane.SetSpec(dataplaneProto)
	return nil
```

Reply:
Yes, I feel that handling this in `PodToDataplane` might be simpler, but it may not address all the cases. I've noticed the issue affecting `Dataplane`, `ZoneIngress`, and `ZoneEgress`. However, given how the data is stored and how the checks are performed, other components like `MeshService` and `MeshGateway` might also be impacted.

Reply:
I don't think so as MeshService doesn't use protobufs. MeshGateway might be have the problem yes

Reply:
Ok, so I will update a previous PR with adjustment to validate if the proto has changed and check if the same is for gateway

Reply:
Closing this in favor of [#11327](https://github.com/kumahq/kuma/pull/11327).